pkg/controller/rdconfig.go

```diff
@@ -21,11 +21,127 @@ import (
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.FieldSelector = fields.Set{"metadata.name": name}.String()
+				obj, err := rdConfigClient.AciV1().RdConfigs(metav1.NamespaceAll).Watch(context.TODO(), options)
+				if err != nil {
+					cont.log.Fatal("Failed to watch RdConfigs during initialization RdConfigsInformer")
+				}
+				return obj, err
+			},
+		})
+}
+
+func (cont *AciController) initRdConfigInformerBase(listWatch *cache.ListWatch) {
+	cont.rdConfigIndexer, cont.rdConfigInformer = cache.NewIndexerInformer(
+		listWatch,
+		&rdConfig.RdConfig{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.RdConfigAdded(obj)
+			},
+			UpdateFunc: func(oldiobj interface{}, newobj interface{}) {
+				cont.RdConfigUpdated(oldiobj, newobj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.RdConfigDeleted(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing RdConfig Informers")
+}
+
+func (cont *AciController) RdConfigAdded(obj interface{}) {
+	rdcon := obj.(*rdConfig.RdConfig)
+	rdconkey, err := cache.MetaNamespaceKeyFunc(rdcon)
+	if err != nil {
+		return
+	}
+	cont.log.Infof("RdConfig: %s added", rdconkey)
+	cont.indexMutex.Lock()
+	cont.rdConfigCache[rdcon.ObjectMeta.Name] = rdcon
+	cont.indexMutex.Unlock()
+	cont.queueRdConfigUpdateByKey(rdconkey)
+}
+
+func (cont *AciController) RdConfigUpdated(oldobj interface{}, newobj interface{}) {
+	oldrdcon := oldobj.(*rdConfig.RdConfig)
+	newrdcon := newobj.(*rdConfig.RdConfig)
+	rdconkey, err := cache.MetaNamespaceKeyFunc(newrdcon)
+	if err != nil {
+		return
+	}
+	if reflect.DeepEqual(oldrdcon.Spec, newrdcon.Spec) {
+		return
+	}
+	cont.log.Infof("Updating RdConfig: %s", rdconkey)
+	cont.indexMutex.Lock()
+	cont.log.Debugf("Updating RdConfig:%s, from: %v\tto: %v", rdconkey, oldrdcon, newrdcon)
+	cont.rdConfigCache[newrdcon.ObjectMeta.Name] = newrdcon
+	cont.indexMutex.Unlock()
+	cont.queueRdConfigUpdateByKey(rdconkey)
+
+}
+
+func (cont *AciController) RdConfigDeleted(obj interface{}) {
+	rdcon := obj.(*rdConfig.RdConfig)
+	rdconkey, err := cache.MetaNamespaceKeyFunc(rdcon)
+	if err != nil {
+		return
+	}
+	cont.indexMutex.Lock()
+	cont.log.Info("Deleting RdConfig object with name: ", rdcon.ObjectMeta.Name)
+	delete(cont.rdConfigCache, rdcon.ObjectMeta.Name)
+	cont.indexMutex.Unlock()
+	cont.queueRdConfigUpdateByKey(rdconkey)
+}
+
+func (cont *AciController) queueRdConfigUpdateByKey(key string) {
+	cont.log.Debug("RdConfig key Queued: ", key)
+	cont.rdConfigQueue.Add(key)
+}
+
+func (cont *AciController) isRdConfigPresent(rdcname string) bool {
+	cont.indexMutex.Lock()
+	_, ok := cont.rdConfigCache[rdcname]
+	cont.indexMutex.Unlock()
+	return ok
+}
+func (cont *AciController) delhandleRdConfig(rdconfig *rdConfig.RdConfig) bool {
+	return cont.syncRdConfig()
+}
+
+func (cont *AciController) handleRdConfig(rdconfig *rdConfig.RdConfig) bool {
```

Comment:
can remove this and set to nil, like other handlers are setting for post delete event.